### PR TITLE
ci: update golangci-lint to latest release

### PR DESCRIPTION
**Pull request by Madhu-1**
This PR updates the  CI tool to the latest release i.e 1.43.0 and fixes all failures which are found in older tools. New tools checks are marked for TODO.

updates #2595 

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

@ceph/ceph-csi-maintainers @ceph/contributors requesting a review on priority as rebasing this kind of PR becomes difficult.

### CODE REVIEW

File: scripts/golangci.yml.in [hunk 1]
```diff
@@ -179,3 +179,10 @@ linters:
     # TODO: enable wrapcheck linter
     # See: https://github.com/ceph/ceph-csi/pull/2268
     - wrapcheck
+    # TODO: enable linters added in golangci-lint 1.43
```
Comment (nixpanic) [inline scripts/golangci.yml.in:182]: have you created a github issue for this?

File: internal/kms/vault_sa.go [hunk 2]
```diff
@@ -320,7 +320,7 @@ func (kms *VaultTenantSA) getTokenPath() (string, error) {
 		return "", err
 	}
 
-	err = os.WriteFile(dir+"/token", []byte(token), 0600)
+	err = os.WriteFile(dir+"/token", []byte(token), 0o600)
```
Comment (nixpanic) [inline internal/kms/vault_sa.go:323]: typo in the commit message title, `0o6000` != `0o600`

File: internal/util/errors.go [hunk 3]
```diff
@@ -39,27 +39,27 @@ var (
 	ErrMissingConfigForMonitor = errors.New("missing configuration of cluster ID for monitor")
 )
 
-type errorPair struct {
+type pairError struct {
```
Comment (nixpanic) [inline internal/util/errors.go:42]: I do not think this is ever used, delete it completely?
Probably a left-over from before the project started to use `%w`.

**Review by nixpanic — CHANGES_REQUESTED** [review]
few minor nits that need addressing

File: scripts/golangci.yml.in (hunk 1 above)
Comment (Madhu-1) [reply scripts/golangci.yml.in:182]: will reuse the existing issue as I have just added an update in the PR description. once PR is merged I will add a link to disabled linters and comment on the issue for tracking.

File: internal/util/errors.go (hunk 3 above)
Comment (Madhu-1) [reply internal/util/errors.go:42]: we are using `JoinErrors` which is a method of  `pairError` in many places.

File: scripts/golangci.yml.in (hunk 1 above)
Comment (Madhu-1) [reply scripts/golangci.yml.in:182]: here is the updated comment https://github.com/ceph/ceph-csi/issues/2595#issuecomment-1018385617

**Review by yati1998 — APPROVED** [review]
LGTM

File: internal/util/errors.go (hunk 3 above)
Comment (nixpanic) [reply internal/util/errors.go:42]: right, I guess that could be replaced as well, but no need to do this in this PR

**Review by nixpanic — APPROVED** [review]

**Comment by nixpanic** [comment]
> /retest ci/centos/mini-e2e/k8s-1.22

Failed to deploy snapshot pieces:
```
Error from server: error when creating "https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/v4.0.0/client/config/crd/snapshot.storage.k8s.io_volumesnapshotcontents.yaml": etcdserver: request timed out
```

[logs](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/mini-e2e_k8s-1.22/detail/mini-e2e_k8s-1.22/1805/pipeline)

**Comment by Madhu-1** [comment]
@mergifyio rebase

**Comment by Madhu-1** [comment]
@mergifyio refresh

**Pull request merged**